ksml/src/main/java/io/axual/ksml/data/mapper/UserTypeStringMapper.java

```diff
@@ -0,0 +1,102 @@
+package io.axual.ksml.data.mapper;
+
+/*-
+ * ========================LICENSE_START=================================
+ * KSML
+ * %%
+ * Copyright (C) 2021 - 2022 Axual B.V.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import io.axual.ksml.data.object.DataBoolean;
+import io.axual.ksml.data.object.DataByte;
+import io.axual.ksml.data.object.DataBytes;
+import io.axual.ksml.data.object.DataDouble;
+import io.axual.ksml.data.object.DataFloat;
+import io.axual.ksml.data.object.DataInteger;
+import io.axual.ksml.data.object.DataLong;
+import io.axual.ksml.data.object.DataNull;
+import io.axual.ksml.data.object.DataShort;
+import io.axual.ksml.data.object.DataString;
+import io.axual.ksml.data.type.DataType;
+import io.axual.ksml.data.type.EnumType;
+import io.axual.ksml.data.type.ListType;
+import io.axual.ksml.data.type.StructType;
+import io.axual.ksml.data.type.TupleType;
+import io.axual.ksml.data.type.UserType;
+import io.axual.ksml.parser.UserTypeParser;
+
+public class UserTypeStringMapper implements StringMapper<UserType> {
+    @Override
+    public String toString(UserType value) {
+        var notation = value.notation().toLowerCase();
+        var type = toString(value.dataType());
+        if (type == null || type.isEmpty()) return notation;
+        return notation + ":" + type;
+    }
+
+    private String toString(DataType value) {
+        if (value == DataNull.DATATYPE) return "null";
```

Comment:
Might be better to move this into some toString() method inside the enum

Reply:
I considered this, but the mapper goes two-way, namely from/to the formal syntax of the UserType. Implementing this in the class itself would mean that the class itself knows its formal syntax. Normally toString() in a class is only used to aid developers or for one-way conversion to String, but not to be parsed back.